### ISSUE TITLE
Thread leak in start_async

### ISSUE THREAD

**Issue by tikh** [slack-ruby/slack-ruby-client#104]
Every time the Slack client is reconnected, a new thread is created. Here is a code to reproduce the issue: https://gist.github.com/tikh/8d43a1a70dca55e89c0645cf7c9f6cc0. Thread count is going up for every reconnect. We are using celluloid-io gem here.

**Comment by dblock** [comment]
The `start_async` method is creating a new actor, via `Actor.new`, we must not be cleaning up the old one correctly.

**Comment by kandadaboggu** [comment]
When you are connecting to few hundred clients the processes runs out of threads very quickly.  In last few days the slack server has been disconnecting connection every few minutes. So this has caused servers die often.

**Comment by dblock** [comment]
Confirming these are Celluloid threads.

```
threads (num = 6): [
  "#<Thread:0x007fd4508bc410>",
  "#<Celluloid::Thread:0x007fd452ff3510>",
  "#<Celluloid::Thread:0x007fd452fe84d0>",
  "#<Celluloid::Thread:0x007fd452fe0f28>",
  "#<Celluloid::Thread:0x007fd452fd3eb8>",
  "#<Celluloid::Thread:0x007fd452b636a8>"
]
```

```
threads (num = 7): [
  "#<Thread:0x007fd4508bc410>",
  "#<Celluloid::Thread:0x007fd452ff3510>",
  "#<Celluloid::Thread:0x007fd452fe84d0>",
  "#<Celluloid::Thread:0x007fd452fe0f28>",
  "#<Celluloid::Thread:0x007fd452fd3eb8>",
  "#<Celluloid::Thread:0x007fd452b636a8>",
  "#<Celluloid::Thread:0x007fd450ae3108>"
]
```

I'm using 

```ruby
      GC.start
      threads = Thread.list
      objects = ObjectSpace.count_objects
      puts "objcount: #{JSON.pretty_generate(objects)}"
      puts "threads (num = #{threads.count}): #{JSON.pretty_generate(threads)}"
```

**Comment by kandadaboggu** [comment]
It looks like we have to call `terminate` on the old actor.
https://github.com/celluloid/celluloid/wiki/Actor-lifecycle

**Comment by dblock** [comment]
Yep, just reached the same conclusion, I am with you. Simpler repro:

```ruby
require 'spec_helper'

RSpec.describe 'reconnect test', skip: (!ENV['SLACK_API_TOKEN'] || !ENV['CONCURRENCY']) && 'missing SLACK_API_TOKEN and/or CONCURRENCY' do
  around do |ex|
    WebMock.allow_net_connect!
    VCR.turned_off { ex.run }
    WebMock.disable_net_connect!
  end

  let(:logger) do
    logger = Logger.new(STDOUT)
    logger.level = Logger::INFO
    logger
  end

  before do
    Thread.abort_on_exception = true

    Slack.configure do |slack|
      slack.logger = logger
    end
  end

  after do
    Slack.config.reset
  end

  let(:client) { Slack::RealTime::Client.new(token: ENV['SLACK_API_TOKEN']) }

  before do
    client.on :hello do
      logger.info "Successfully connected, welcome '#{client.self.name}' to the '#{client.team.name}' team at https://#{client.team.domain}.slack.com."

      @stopping = true
      client.stop!
    end

    client.on :close do
      if @stopping
        @stopping = false
        next
      end

      GC.start
      threads = Thread.list
      objects = ObjectSpace.count_objects
      puts "objcount: #{JSON.pretty_generate(objects)}"
      puts "threads (num = #{threads.count}): #{JSON.pretty_generate(threads)}"

      logger.info 'Closing, and restarting ...'
      sleep 10
      client.start_async
      # TODO: restart
    end
  end

  it 'reconnects in a loop' do
    client.start_async

    loop do
      sleep 1
    end
  end
end
```

**Comment by dblock** [comment]
Check out https://github.com/slack-ruby/slack-ruby-client/pull/105. LMK if you think I am missing anything.

**Comment by dblock** [comment]
Fun graph, before:

![screen shot 2016-08-26 at 1 06 55 pm](https://cloud.githubusercontent.com/assets/542335/18013844/08d927a6-6b8e-11e6-992c-2265a4df55f1.png)

**Comment by dblock** [comment]
After the fix:

![screen shot 2016-08-26 at 1 08 04 pm](https://cloud.githubusercontent.com/assets/542335/18013870/28ffc684-6b8e-11e6-9a2b-f90e58e81bd0.png)

**Comment by tikh** [comment]
Thank you @dblock and @kandadaboggu!

**Comment by dblock** [comment]
This was released in 0.7.7.
